hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java

```diff
@@ -0,0 +1,126 @@
+        this.addressBookValidator =
+                requireNonNull(addressBookValidator, "The supplied argument 'addressBookValidator' must not be null");
+    }
+
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final var op = txn.nodeCreate();
+        final var accountId = validateAccountID(op.accountIdOrElse(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(op.gossipEndpoint().isEmpty(), INVALID_GOSSIP_ENDPOINT);
+        validateFalsePreCheck(op.serviceEndpoint().isEmpty(), INVALID_SERVICE_ENDPOINT);
+        validateFalsePreCheck(op.gossipCaCertificate().equals(Bytes.EMPTY), INVALID_GOSSIP_CAE_CERTIFICATE);
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) {
+        requireNonNull(context);
+    }
+
+    @Override
+    public void handle(@NonNull final HandleContext handleContext) {
+        requireNonNull(handleContext);
+
+        final var op = handleContext.body().nodeCreate();
+
+        final var configuration = handleContext.configuration();
+        final var nodeConfig = configuration.getConfigData(NodesConfig.class);
+        final var nodeStore = handleContext.writableStore(WritableNodeStore.class);
+        final var accountStore = handleContext.readableStore(ReadableAccountStore.class);
+        final var accountId = op.accountIdOrElse(AccountID.DEFAULT);
+
+        validateFalse(nodeStore.sizeOfState() >= nodeConfig.maxNumber(), MAX_NODES_CREATED);
+        validateTrue(accountStore.contains(accountId), INVALID_NODE_ACCOUNT_ID);
+        addressBookValidator.validateDescription(op.description(), nodeConfig);
+        addressBookValidator.validateGossipEndpoint(op.gossipEndpoint(), nodeConfig);
+        addressBookValidator.validateServiceEndpoint(op.serviceEndpoint(), nodeConfig);
+
+        final var nodeBuilder = new Node.Builder()
+                .accountId(op.accountId())
+                .description(op.description())
+                .gossipEndpoint(op.gossipEndpoint())
+                .serviceEndpoint(op.serviceEndpoint())
+                .gossipEndpoint(op.gossipEndpoint())
+                .serviceEndpoint(op.serviceEndpoint())
+                .gossipCaCertificate(op.gossipCaCertificate())
+                .grpcCertificateHash(op.grpcCertificateHash());
+        final var node = nodeBuilder.nodeId(getNextNodeID(nodeStore)).build();
+
+        nodeStore.put(node);
+
+        final var recordBuilder = handleContext.recordBuilder(NodeCreateRecordBuilder.class);
+
+        recordBuilder.nodeID(node.nodeId());
+    }
+
+    private long getNextNodeID(WritableNodeStore nodeStore) {
+        final var nodeIds = nodeStore.keys();
+        long max = 0;
+        while (nodeIds.hasNext()) {
+            max = Math.max(max, nodeIds.next().number());
+        }
```

Comment:
We can use Streams API for it to look concise

Reply:
Will change it in the next PR.